File: components/src/main/java/com/omvp/components/NoticeDialogComponentView.java
package com.omvp.components;

import android.content.Context;
import android.support.v7.widget.AppCompatTextView;
import android.util.AttributeSet;
import android.view.View;

import butterknife.ButterKnife;

public class NoticeDialogComponentView extends BaseComponentView {

    AppCompatTextView mTitleTextView;
    AppCompatTextView mDescriptionTextView;
    AppCompatTextView mAcceptTextView;
    AppCompatTextView mDenyTextView;

    public NoticeDialogComponentView(Context context) {
        super(context);
    }

    public NoticeDialogComponentView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public NoticeDialogComponentView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    protected void loadAttributes(Context context, AttributeSet attrs) {

    }

    @Override
    protected void bindViews() {
        mTitleTextView = ButterKnife.findById(this, R.id.title);
        mDescriptionTextView = ButterKnife.findById(this, R.id.description);
        mAcceptTextView = ButterKnife.findById(this, R.id.accept);
        mDenyTextView = ButterKnife.findById(this, R.id.deny);
    }

    @Override
    protected void loadData() {

    }

    @Override
    protected int getLayoutId() {
        return R.layout.notice_dialog_component_view;
    }

    public void setTitleText(String text) {
        mTitleTextView.setText(text);
    }

    public void setDescriptionText(String text) {
        mDescriptionTextView.setText(text);
    }

    public void setAcceptTextButton(String text, OnClickListener onClickListener) {
        mAcceptTextView.setText(text);
        mAcceptTextView.setOnClickListener(onClickListener);
        mAcceptTextView.setVisibility(View.VISIBLE);
    }

    public void setDenyTextButton(String text, OnClickListener onClickListener) {
        mDenyTextView.setText(text);
        mDenyTextView.setOnClickListener(onClickListener);
        mDenyTextView.setVisibility(View.VISIBLE);
    }

}

File: components/src/main/java/com/omvp/components/NoticeDialogComponentView.kt
package com.omvp.components

import android.content.Context
import android.util.AttributeSet
import android.view.View
import androidx.appcompat.widget.AppCompatTextView

class NoticeDialogComponentView : BaseComponentView {

    private lateinit var mTitleTextView: AppCompatTextView
    private lateinit var mDescriptionTextView: AppCompatTextView
    private lateinit var mAcceptTextView: AppCompatTextView
    private lateinit var mDenyTextView: AppCompatTextView

    override val layoutId: Int
        get() = R.layout.notice_dialog_component_view

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    override fun loadAttributes(context: Context, attrs: AttributeSet?) {

    }

    override fun bindViews() {
        mTitleTextView = findViewById(R.id.title)
        mDescriptionTextView = findViewById(R.id.description)
        mAcceptTextView = findViewById(R.id.accept)
        mDenyTextView = findViewById(R.id.deny)
    }

    override fun loadData() {

    }

    fun setTitleText(text: String) {
        mTitleTextView.text = text
    }

    fun setDescriptionText(text: String) {
        mDescriptionTextView.text = text
    }

    fun setAcceptTextButton(text: String, onClickListener: View.OnClickListener) {
        mAcceptTextView.text = text
        mAcceptTextView.setOnClickListener(onClickListener)
        mAcceptTextView.visibility = View.VISIBLE
    }

    fun setDenyTextButton(text: String, onClickListener: View.OnClickListener) {
        mDenyTextView.text = text
        mDenyTextView.setOnClickListener(onClickListener)
        mDenyTextView.visibility = View.VISIBLE
    }
}
